From bf559363f7f26afdd3b3ce5b6d9a57f8427b450b Mon Sep 17 00:00:00 2001
From: David Reaver <me@davidreaver.com>
Date: Wed, 29 Jan 2025 14:52:26 -0800
Subject: [PATCH 0/2] *** SUBJECT HERE ***
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
    Rafael J. Wysocki <rafael@kernel.org>,
    Danilo Krummrich <dakr@kernel.org>
Cc: Steven Rostedt <rostedt@goodmis.org>,
    Christian Brauner <brauner@kernel.org>,
    linux-fsdevel@vger.kernel.org,
    linux-kernel@vger.kernel.org

*** BLURB HERE ***

David Reaver (2):
  debugfs: Use opaque debugfs_node handle to support kernfs migration
  debugfs: tmp commit to show coccinelle output

 arch/powerpc/kernel/iommu.c                   |   2 +-
 arch/powerpc/kernel/kdebugfs.c                |   2 +-
 arch/powerpc/kernel/traps.c                   |   2 +-
 arch/powerpc/kvm/book3s_hv.c                  |   6 +-
 arch/powerpc/kvm/book3s_xics.h                |   2 +-
 arch/powerpc/kvm/book3s_xive.h                |   2 +-
 arch/powerpc/kvm/powerpc.c                    |   3 +-
 arch/powerpc/kvm/timing.c                     |   2 +-
 arch/powerpc/kvm/timing.h                     |   4 +-
 arch/powerpc/platforms/powernv/memtrace.c     |   4 +-
 arch/powerpc/platforms/powernv/opal-imc.c     |   4 +-
 arch/powerpc/platforms/powernv/opal-lpc.c     |   4 +-
 arch/powerpc/platforms/powernv/opal-xscom.c   |   4 +-
 arch/powerpc/platforms/powernv/vas-debug.c    |   6 +-
 arch/powerpc/platforms/powernv/vas.h          |   2 +-
 arch/powerpc/platforms/pseries/dtl.c          |   2 +-
 arch/powerpc/platforms/pseries/hvCall_inst.c  |   2 +-
 arch/powerpc/platforms/pseries/lpar.c         |   2 +-
 arch/powerpc/sysdev/xive/common.c             |   4 +-
 arch/s390/hypfs/hypfs.h                       |   2 +-
 arch/s390/hypfs/hypfs_dbfs.c                  |   2 +-
 arch/s390/kernel/debug.c                      |   4 +-
 arch/s390/kernel/hiperdispatch.c              |   2 +-
 arch/s390/kernel/kdebugfs.c                   |   2 +-
 arch/s390/kernel/sysinfo.c                    |   2 +-
 arch/s390/kernel/wti.c                        |   2 +-
 arch/s390/pci/pci_debug.c                     |   2 +-
 arch/x86/kernel/callthunks.c                  |   2 +-
 arch/x86/kernel/cpu/mce/core.c                |   6 +-
 arch/x86/kernel/cpu/mce/inject.c              |   2 +-
 arch/x86/kernel/cpu/mce/internal.h            |   2 +-
 arch/x86/kernel/cpu/mce/severity.c            |   2 +-
 arch/x86/kernel/cpu/resctrl/internal.h        |   4 +-
 arch/x86/kernel/cpu/resctrl/rdtgroup.c        |   2 +-
 arch/x86/kernel/itmt.c                        |   2 +-
 arch/x86/kernel/kdebugfs.c                    |  12 +-
 arch/x86/kvm/debugfs.c                        |   3 +-
 arch/x86/mm/debug_pagetables.c                |   2 +-
 arch/x86/platform/atom/punit_atom_debug.c     |   2 +-
 arch/x86/platform/intel/iosf_mbi.c            |   2 +-
 arch/x86/xen/debugfs.c                        |   4 +-
 arch/x86/xen/p2m.c                            |   2 +-
 arch/x86/xen/xen-ops.h                        |   2 +-
 block/blk-core.c                              |   2 +-
 block/blk-mq-debugfs.c                        |   4 +-
 block/blk-rq-qos.h                            |   2 +-
 block/blk.h                                   |   2 +-
 drivers/acpi/acpi_dbg.c                       |   2 +-
 drivers/acpi/apei/apei-base.c                 |   4 +-
 drivers/acpi/apei/apei-internal.h             |   2 +-
 drivers/acpi/apei/einj-core.c                 |   2 +-
 drivers/acpi/debugfs.c                        |   2 +-
 drivers/acpi/ec_sys.c                         |   4 +-
 drivers/acpi/internal.h                       |   2 +-
 drivers/base/component.c                      |   2 +-
 drivers/base/regmap/internal.h                |   2 +-
 drivers/base/regmap/regmap-debugfs.c          |   2 +-
 drivers/bus/mhi/host/debugfs.c                |   2 +-
 drivers/bus/mvebu-mbus.c                      |   6 +-
 drivers/dma/bcm-sba-raid.c                    |   2 +-
 drivers/dma/dmaengine.c                       |   2 +-
 drivers/dma/dmaengine.h                       |   4 +-
 drivers/dma/dw-edma/dw-edma-v0-debugfs.c      |  11 +-
 drivers/dma/dw-edma/dw-hdma-v0-debugfs.c      |  13 +-
 drivers/dma/hisi_dma.c                        |   2 +-
 drivers/dma/idxd/debugfs.c                    |   2 +-
 drivers/dma/idxd/idxd.h                       |   4 +-
 drivers/dma/ptdma/ptdma-debugfs.c             |   2 +-
 drivers/dma/pxa_dma.c                         |   6 +-
 drivers/dma/qcom/hidma.h                      |   2 +-
 drivers/dma/qcom/hidma_dbg.c                  |   2 +-
 drivers/dma/xilinx/xilinx_dpdma.c             |   2 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_aca.c       |   3 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_aca.h       |   3 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_debugfs.c   |   2 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_mca.c       |   3 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_mca.h       |   3 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c       |   8 +-
 drivers/gpu/drm/amd/amdgpu/amdgpu_ras.h       |   2 +-
 drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c      |   2 +-
 .../amd/display/amdgpu_dm/amdgpu_dm_debugfs.c |   2 +-
 drivers/gpu/drm/bridge/ite-it6505.c           |   2 +-
 drivers/gpu/drm/bridge/panel.c                |   2 +-
 drivers/gpu/drm/bridge/synopsys/dw-mipi-dsi.c |   2 +-
 drivers/gpu/drm/bridge/ti-sn65dsi86.c         |   2 +-
 .../gpu/drm/display/drm_bridge_connector.c    |   2 +-
 drivers/gpu/drm/drm_debugfs.c                 |  20 +-
 drivers/gpu/drm/drm_debugfs_crc.c             |   2 +-
 drivers/gpu/drm/drm_drv.c                     |   2 +-
 drivers/gpu/drm/drm_internal.h                |   4 +-
 .../display/intel_display_debugfs_params.c    |  10 +-
 drivers/gpu/drm/i915/display/intel_fbc.c      |   2 +-
 drivers/gpu/drm/i915/gt/intel_gt_debugfs.c    |   7 +-
 drivers/gpu/drm/i915/gt/intel_gt_debugfs.h    |   2 +-
 .../drm/i915/gt/intel_gt_engines_debugfs.c    |   3 +-
 .../drm/i915/gt/intel_gt_engines_debugfs.h    |   3 +-
 drivers/gpu/drm/i915/gt/intel_gt_pm_debugfs.c |   3 +-
 drivers/gpu/drm/i915/gt/intel_gt_pm_debugfs.h |   3 +-
 drivers/gpu/drm/i915/gt/intel_sseu_debugfs.c  |   3 +-
 drivers/gpu/drm/i915/gt/intel_sseu_debugfs.h  |   3 +-
 .../gpu/drm/i915/gt/uc/intel_gsc_uc_debugfs.c |   3 +-
 .../gpu/drm/i915/gt/uc/intel_gsc_uc_debugfs.h |   3 +-
 drivers/gpu/drm/i915/gt/uc/intel_guc.h        |   2 +-
 .../gpu/drm/i915/gt/uc/intel_guc_debugfs.c    |   3 +-
 .../gpu/drm/i915/gt/uc/intel_guc_debugfs.h    |   3 +-
 drivers/gpu/drm/i915/gt/uc/intel_guc_log.c    |   2 +-
 .../drm/i915/gt/uc/intel_guc_log_debugfs.c    |   2 +-
 .../drm/i915/gt/uc/intel_guc_log_debugfs.h    |   2 +-
 .../gpu/drm/i915/gt/uc/intel_huc_debugfs.c    |   3 +-
 .../gpu/drm/i915/gt/uc/intel_huc_debugfs.h    |   3 +-
 drivers/gpu/drm/i915/gt/uc/intel_uc_debugfs.c |   5 +-
 drivers/gpu/drm/i915/gt/uc/intel_uc_debugfs.h |   3 +-
 drivers/gpu/drm/i915/i915_debugfs_params.c    |  16 +-
 drivers/gpu/drm/i915/i915_debugfs_params.h    |   2 +-
 drivers/gpu/drm/i915/pxp/intel_pxp_debugfs.c  |   2 +-
 drivers/gpu/drm/imagination/pvr_debugfs.c     |   2 +-
 drivers/gpu/drm/imagination/pvr_fw_trace.c    |   3 +-
 drivers/gpu/drm/imagination/pvr_fw_trace.h    |   3 +-
 drivers/gpu/drm/imagination/pvr_params.c      |   2 +-
 drivers/gpu/drm/imagination/pvr_params.h      |   3 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_core_irq.h  |   2 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_core_perf.c |   5 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_core_perf.h |   3 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_encoder.c   |   3 +-
 .../gpu/drm/msm/disp/dpu1/dpu_hw_interrupts.c |   2 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_hw_sspp.c   |   4 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_hw_sspp.h   |   2 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_kms.c       |   7 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_vbif.c      |   3 +-
 drivers/gpu/drm/msm/disp/dpu1/dpu_vbif.h      |   3 +-
 drivers/gpu/drm/msm/dp/dp_display.c           |   3 +-
 drivers/gpu/drm/msm/dp/dp_display.h           |   3 +-
 drivers/gpu/drm/msm/dp/dp_drm.c               |   6 +-
 drivers/gpu/drm/msm/msm_debugfs.c             |   2 +-
 drivers/gpu/drm/nouveau/dispnv50/crc.c        |   2 +-
 drivers/gpu/drm/nouveau/nouveau_debugfs.c     |   4 +-
 drivers/gpu/drm/nouveau/nouveau_debugfs.h     |   2 +-
 drivers/gpu/drm/nouveau/nouveau_drm.c         |   2 +-
 .../gpu/drm/nouveau/nvkm/subdev/gsp/r535.c    |  10 +-
 drivers/gpu/drm/omapdrm/dss/dss.c             |   4 +-
 drivers/gpu/drm/panel/panel-edp.c             |   3 +-
 drivers/gpu/drm/panel/panel-sitronix-st7703.c |   2 +-
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c  |   2 +-
 drivers/gpu/drm/tegra/dc.c                    |   4 +-
 drivers/gpu/drm/ttm/ttm_device.c              |   2 +-
 drivers/gpu/drm/ttm/ttm_module.h              |   2 +-
 drivers/gpu/drm/ttm/ttm_resource.c            |   2 +-
 drivers/gpu/drm/xe/xe_gsc_debugfs.c           |   2 +-
 drivers/gpu/drm/xe/xe_gsc_debugfs.h           |   2 +-
 drivers/gpu/drm/xe/xe_gt_debugfs.c            |   2 +-
 drivers/gpu/drm/xe/xe_gt_sriov_pf_debugfs.c   |   7 +-
 drivers/gpu/drm/xe/xe_gt_sriov_pf_debugfs.h   |   6 +-
 drivers/gpu/drm/xe/xe_gt_sriov_vf_debugfs.c   |   5 +-
 drivers/gpu/drm/xe/xe_gt_sriov_vf_debugfs.h   |   3 +-
 drivers/gpu/drm/xe/xe_guc_debugfs.c           |   2 +-
 drivers/gpu/drm/xe/xe_guc_debugfs.h           |   2 +-
 drivers/gpu/drm/xe/xe_huc_debugfs.c           |   2 +-
 drivers/gpu/drm/xe/xe_huc_debugfs.h           |   2 +-
 drivers/gpu/drm/xe/xe_uc_debugfs.c            |   4 +-
 drivers/gpu/drm/xe/xe_uc_debugfs.h            |   2 +-
 drivers/gpu/drm/xlnx/zynqmp_dp.c              |   4 +-
 drivers/gpu/host1x/dev.h                      |   2 +-
 drivers/gpu/vga/vga_switcheroo.c              |   2 +-
 drivers/misc/cxl/cxl.h                        |  30 +--
 drivers/misc/cxl/debugfs.c                    |  21 +-
 drivers/misc/eeprom/idt_89hpesx.c             |   4 +-
 drivers/misc/genwqe/card_base.c               |   2 +-
 drivers/misc/genwqe/card_base.h               |   4 +-
 drivers/misc/genwqe/card_debugfs.c            |   2 +-
 drivers/misc/lkdtm/core.c                     |   2 +-
 drivers/misc/mei/debugfs.c                    |   2 +-
 drivers/misc/mei/mei_dev.h                    |   2 +-
 drivers/misc/xilinx_tmr_inject.c              |   4 +-
 drivers/phy/phy-core.c                        |   2 +-
 drivers/phy/realtek/phy-rtk-usb2.c            |   4 +-
 drivers/phy/realtek/phy-rtk-usb3.c            |   4 +-
 drivers/pinctrl/core.c                        |   4 +-
 drivers/pinctrl/core.h                        |   2 +-
 drivers/pinctrl/pinconf.c                     |   4 +-
 drivers/pinctrl/pinconf.h                     |   4 +-
 drivers/pinctrl/pinmux.c                      |   4 +-
 drivers/pinctrl/pinmux.h                      |   4 +-
 fs/debugfs/file.c                             | 192 +++++++++--------
 fs/debugfs/inode.c                            | 142 ++++++++-----
 include/drm/drm_connector.h                   |   2 +-
 include/drm/drm_crtc.h                        |   2 +-
 include/drm/drm_debugfs.h                     |  13 +-
 include/drm/drm_device.h                      |   2 +-
 include/drm/drm_drv.h                         |   5 +-
 include/drm/drm_encoder.h                     |   2 +-
 include/drm/drm_file.h                        |   4 +-
 include/drm/ttm/ttm_resource.h                |   2 +-
 include/kunit/test.h                          |   2 +-
 include/linux/blkdev.h                        |   6 +-
 include/linux/debugfs.h                       | 194 ++++++++++--------
 include/linux/dmaengine.h                     |   2 +-
 include/linux/fault-inject.h                  |  10 +-
 include/linux/i2c.h                           |   4 +-
 include/linux/iommu.h                         |   2 +-
 include/linux/kvm_host.h                      |   5 +-
 include/linux/mhi.h                           |   2 +-
 include/linux/moxtet.h                        |   2 +-
 include/linux/phy/phy.h                       |   2 +-
 include/linux/proc_fs.h                       |   4 +-
 include/linux/shrinker.h                      |   2 +-
 include/linux/usb.h                           |   2 +-
 include/net/cfg80211.h                        |   2 +-
 include/sound/core.h                          |   4 +-
 lib/debugobjects.c                            |   2 +-
 lib/dynamic_debug.c                           |   2 +-
 lib/fault-inject.c                            |  13 +-
 lib/kunit/debugfs.c                           |   2 +-
 lib/memory-notifier-error-inject.c            |   2 +-
 lib/netdev-notifier-error-inject.c            |   2 +-
 lib/notifier-error-inject.c                   |  12 +-
 lib/notifier-error-inject.h                   |   2 +-
 lib/of-reconfig-notifier-error-inject.c       |   2 +-
 lib/pm-notifier-error-inject.c                |   2 +-
 lib/stackdepot.c                              |   2 +-
 lib/test_fpu_glue.c                           |   2 +-
 mm/backing-dev.c                              |   2 +-
 mm/cma_debug.c                                |   7 +-
 mm/fail_page_alloc.c                          |   2 +-
 mm/failslab.c                                 |   2 +-
 mm/hwpoison-inject.c                          |   2 +-
 mm/internal.h                                 |   8 +-
 mm/kfence/core.c                              |   2 +-
 mm/page_owner.c                               |   2 +-
 mm/shrinker_debug.c                           |  11 +-
 mm/slub.c                                     |   4 +-
 mm/vmstat.c                                   |   2 +-
 mm/zsmalloc.c                                 |   4 +-
 mm/zswap.c                                    |   2 +-
 net/wireless/core.c                           |   2 +-
 net/wireless/debugfs.c                        |   4 +-
 sound/core/jack.c                             |   2 +-
 sound/core/sound.c                            |   2 +-
 sound/drivers/pcmtest.c                       |   2 +-
 virt/kvm/kvm_main.c                           |  11 +-
 239 files changed, 756 insertions(+), 612 deletions(-)


base-commit: 05dbaf8dd8bf537d4b4eb3115ab42a5fb40ff1f5
